### ISSUE TITLE
Numerically stable OLS

### ISSUE THREAD

**Issue by jseabold** [rasbt/mlxtend#56]
You might consider switching to using SVD (pseudoinverse) or QR decomposition for OLS. For SVD you could update after seeing new data. R's `lm` uses QR by default for some nice features you get for ANOVA after.

https://github.com/statsmodels/statsmodels/blob/master/statsmodels/regression/linear_model.py#L185

Translating from weird statsmodels notation

For SVD

```
beta = np.linalg.pinv(X).dot(y)
```

For QR

```
Q, R = np.linalg.qr(X)
# for ANOVA
effects = np.dot(Q.T, y)
beta = np.linalg.solve(R, effects)
```

**Comment by rasbt** [comment]
Sounds like a good idea, thanks for the suggestion. I am trying to take a look at it soon!

**Comment by PuneetGrov3r** [comment]
I have added this functionality in local and added tests for it too.
This is my first time contributing, where should I add pull request to? master?

**Comment by rasbt** [comment]
Sounds great, thanks. It would be best to make the changes in a new feature branch, For example

```bash
git checkout -b stable-ols
# add changes
git add .
git commit -m 'numerically stable ols
git push origin stable-ols
```

and then initiate the PR via the web interface.

**Comment by PuneetGrov3r** [comment]
Thanks @rasbt

**Comment by PuneetGrov3r** [comment]
I have made a pull request, #575 
Few things about the change:
1. Added a new parameter to LinearRegression, method, which defaults to sgd. Now can be 'sgd', 'qr', 'svd' or 'direct'.
1. Existing codes using direct analytical method will break, because that you have to use method='direct'.

**Comment by rasbt** [comment]
Fixed in #575
